### ISSUE TITLE
Ehnancing the html output of asciidoctor using "type" attributes, so that pandoc better understands it and give better conversions

### ISSUE THREAD

**Issue by zaxebo1** [asciidoctor/asciidoctor#1414]
Note that curently pandoc does not take asciidoc as input format, the work is going on.

> There was a discussion at ( https://github.com/jgm/pandoc/issues/1456#issuecomment-118644517 ) on converting generated html (from asciidoctor) as input format to pandoc
>
> Then the jgm( writer of pandoc John MacFarlane) suggested at https://github.com/jgm/pandoc/issues/1456#issuecomment-118644600
>>  "Or maybe asciidoctor could be persuaded to insert the needed type attributes in the HTML."

 
 I am writing this comment of the pandoc author here, so that may be you will like to get in touch with jgm(writer of pandoc) regarding enhancing the html5 output of asciidoctor  
 ( https://github.com/jgm/pandoc/issues/1456#issuecomment-118644600 ), so that pandoc has better understanding of that html as input format, and pandoc gives us better conversions.

**Comment by zaxebo1** [comment]
I am reproducing the expectation of pandoc in detail here from https://github.com/jgm/pandoc/issues/1456#issuecomment-118644517

> Depending on how asciidoc formats the notes, you may be able to get the HTML reader to parse them as notes. If you use -f html+epub_html_exts, then pandoc will interpret an element with the type attribute set to footnote or rearnote as a note, and an element with the type attribute set to noteref as a note reference, where the href attribute is an internal link to the corresponding footnote or rearnote. It looks as if asciidoc doesn't quite do it that way, but you could use a filter to add the needed type attributes, and then you'd be there.

This^ is the expectation of JohnMacFarlane  from html output of asciidoctor

**Comment by mojavelinux** [comment]
This is related to a larger issue, #242.

**Comment by mojavelinux** [comment]
Of course, you can customize absolutely any of the HTML that Asciidoctor generates using custom templates or a custom converter. What Asciidoctor generates by default is just that, a default.

For instance, in Asciidoctor EPUB3, we output a completely different set of HTML that is much more semantic than what Asciidoctor currently produces. It's also a prototype for the HTML we want to eventually adopt in Asciidoctor core.

Templates to hack: https://github.com/asciidoctor/asciidoctor-backends/tree/master/slim/html5

EPUB3 converter: https://github.com/asciidoctor/asciidoctor-epub3/blob/master/lib/asciidoctor-epub3/converter.rb

**Comment by zaxebo1** [comment]
Thanks for the pointers in right direction

**Comment by mojavelinux** [comment]
:+1:
